google/field_helpers.go

```diff
@@ -0,0 +1,40 @@
+package google
+
+import (
+	"fmt"
+	"regexp"
+)
+
+const networkLinkTemplate = "projects/%s/global/networks/%s"
+
+var networkLinkRegex = regexp.MustCompile(fmt.Sprintf(networkLinkTemplate, "(.+)", "(.+)"))
+
+type NetworkFieldValue struct {
+	Project string
+	Name    string
+}
+
+// Parses a `network` supporting 4 different formats:
+// - https://www.googleapis.com/compute/{version}/projects/myproject/global/networks/my-network
+// - projects/myproject/global/networks/my-network
+// - global/networks/my-network (default project is used)
+// - my-network (default project is used)
+func ParseNetworkFieldValue(network string, config *Config) *NetworkFieldValue {
+	if networkLinkRegex.MatchString(network) {
+		parts := networkLinkRegex.FindStringSubmatch(network)
+
+		return &NetworkFieldValue{
+			Project: parts[1],
+			Name:    parts[2],
+		}
+	}
+
+	return &NetworkFieldValue{
+		Project: config.Project,
+		Name:    GetResourceNameFromSelfLink(network),
+	}
+}
+
+func (f *NetworkFieldValue) RelativeLink() string {
```

Comment:
Minor quibble; if the struct is small it's nice to not use a ptr here for the receiving struct; that way its obvious that your method doesn't mutate the struct its called on (there's runtime benefits as well but I don't think we should be concerned with that)

Reply:
Done.